src/realm/util/thread.hpp

```diff
@@ -41,6 +41,7 @@
 #include <realm/util/assert.hpp>
 #include <realm/util/terminate.hpp>
 #include <memory>
+#include <stdexcept>
```

Comment:
This is needed for `std::runtime_exception`. I guess we were getting this implicitly from some other header. One of the items in https://gcc.gnu.org/gcc-10/changes.html is 
> Reduced header dependencies, leading to faster compilation for some code.